inspirehep/modules/orcid/schema.py

```diff
@@ -0,0 +1,148 @@
+# or submit itself to any jurisdiction.
+
+import dojson
+
+from dojson import utils
+
+from inspirehep.utils.record import get_abstract, get_subtitle, get_title
+
+orcid_overdo = dojson.Overdo()
+
+
+@orcid_overdo.over('title', 'titles')
+def title_rule(self, key, value):
+    title = get_title({"titles": value})
+    if title == '':
+        raise KeyError
+    subtitle = ""
+    try:
+        subtitle = get_subtitle({"titles": value})
+    except KeyError:
+        pass
+    return {"title": title,
+            "subtitle": subtitle}
+
+
+@orcid_overdo.over('journal_title', 'publication_info')
+def publication_rule(self, key, value):
+    try:
+        return value[0].get('journal_title')
+    except (TypeError, KeyError):
+        pass
+
+
+@orcid_overdo.over('short_description', 'abstracts')
+def abstract_rule(self, key, value):
+    try:
+        return get_abstract({"abstracts": value})
+    except (TypeError, KeyError):
+        pass
+
+
+@orcid_overdo.over('publication_date', 'imprints')
+def date_rule(self, key, value):
+    try:
+        date = value[0].get('date')
+        date = date.split('-')
+        final_date = ['', '', '']
+        if date[0]:
+            final_date[0] = date[0]
+        if date[1]:
+            final_date[1] = date[1]
+        if date[2]:
+            final_date[2] = date[2]
+        publication_date = {'year': int(final_date[0]),
+                            'month': int(final_date[1]),
+                            'day': int(final_date[2])
+                            }
+        return publication_date
+    except (TypeError, IndexError):
+        pass
+
+
+@orcid_overdo.over('work_external_identifier', 'arxiv_eprints|dois')
+@dojson.utils.for_each_value
+def external_id_rule(self, key, value):
+    if key == 'dois':
+        return{
+            'external-identifier-type': 'DOI',
+            'external-identifier-id': value.get('value')
+        }
+    if key == 'arxiv_eprints':
+        return {
+            'external-identifier-type': 'ARXIV',
+            'external-identifier-id': value.get('value')
+        }
+
+
+@orcid_overdo.over('contributors', 'authors')
+def authors_rule(self, key, value):
+    value = utils.force_list(value)
+    orcid_authors = []
+    try:
+        orcid_authors.append({
+            'credit-name': value[0].get('full_name'),
+            'contributor-orcid': value[0].get('orcid') if 'orcid' in value[0] else '',
+            'contributor-attributes': {
+                'contributor-role': 'AUTHOR',
+                'contributor-sequence': 'FIRST'
+            }
+        })
+        for author in value[1:]:
+            orcid_authors.append({
+                'credit-name': author.get('full_name'),
+                'contributor-orcid': author.get('orcid') if 'orcid' in author else '',
+                'contributor-attributes': {
+                    'contributor-role': 'AUTHOR',
+                    'contributor-sequence': 'ADDITIONAL'
```

Comment:
Let's see what ORCID people say about this. I remember we had some discussions in the past about the size of our giant authorlists.